jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererTelemetryService.kt

```diff
@@ -218,6 +220,42 @@ class CodeWhispererTelemetryService {
             credentialStartUrl = startUrl
         )
     }
+    fun sendClientComponentLatencyEvent(states: InvocationContext) {
+        val requestContext = states.requestContext
+        val responseContext = states.responseContext
+        val codewhispererLanguage = requestContext.fileContextInfo.programmingLanguage.toTelemetryType()
+        val startUrl = getConnectionStartUrl(requestContext.connection)
+        CodewhispererTelemetry.clientComponentLatency(
+            project = requestContext.project,
+            codewhispererSessionId = responseContext.sessionId,
+            codewhispererRequestId = requestContext.latencyContext.firstRequestId,
+            codewhispererFirstCompletionLatency = requestContext.latencyContext.paginationFirstCompletionTime,
+            codewhispererPreprocessingLatency = TimeUnit.NANOSECONDS.toMillis(
+                requestContext.latencyContext.codewhispererPreprocessingEnd -
+                    requestContext.latencyContext.codewhispererPreprocessingStart
+            ).toDouble(),
+            codewhispererEndToEndLatency = TimeUnit.NANOSECONDS.toMillis(
+                requestContext.latencyContext.codewhispererEndToEndEnd -
+                    requestContext.latencyContext.codewhispererEndToEndStart
+            ).toDouble(),
```

Comment:
nit: adding some extension functions off `LatencyContext` for computing times might help readability slightly. Something like `getCodewhispererEndToEndTime` maybe?

Reply:
make sense